File: FizzBuzz/FizzBuzz.cpp
#include <iostream>

class FizzBuzz {
    private:
        const int FIZZ = 3;
        const int BUZZ = 5;
        int number = 0;

        bool isFizz(const int num) {
            return num % FIZZ == 0;
        }

        bool isBuzz(const int num) {
            return num % BUZZ == 0;
        }

    public:
        FizzBuzz(const int num) {
            number = num;
        }

        void fizz_buzz() {
            for (int i = 1; i <= number; i++) {
                if(isFizz(i) && isBuzz(i)) {
                    std::cout << "FizzBuzz";
                } else if (isFizz(i)) {
                    std::cout << "Fizz";
                } else if (isBuzz(i)) {
                    std::cout << "Buzz";
                } else {
                    std::cout << i;
                }
                std::cout << "\n";
            }
            return;
        }
};

int main(void) {
    FizzBuzz fz(100);
    fz.fizz_buzz();
    return 0;
}

File: FizzBuzz/FizzBuzz.java
class FizzBuzz {

    private static final int FIZZ_NUMBER = 3;
    private static final int BUZZ_NUMBER = 5;
    public static void main(String[] args) {
        int num = 100;

        for (int i = 1; i <= num; i++) {
            if (isFizz(i) && isBuzz(i)) {
                System.out.println("FizzBuzz");
            } else if (isFizz(i)) {
                System.out.println("Fizz");
            } else if (isBuzz(i)) {
                System.out.println("Buzz");
            } else {
                System.out.println(i);
            }
        }
    }
    private static boolean isFizz(final int NUMBER) {
        return NUMBER % FIZZ_NUMBER == 0;
    }

    private static boolean isBuzz(final int NUMBER) {
        return NUMBER % BUZZ_NUMBER == 0;
    }
}